lib/Dialect/RTL/Ops.cpp

```diff
@@ -192,70 +218,172 @@ OpFoldResult OrOp::fold(ArrayRef<Attribute> operands) {
   if (matchPattern(inputs()[size - 1], m_RConstant(value)) &&
       value.isAllOnesValue())
     return inputs()[size - 1];
-
-  /// TODO: or(..., 0) -> or(...) -- identity
-  /// TODO: or(..., x, x) -> or(..., x) -- idempotent
-  /// TODO: or(..., c1, c2) -> or(..., c3) where c3 = c1 | c2 -- constant
-  /// folding
-  /// TODO: or(x, or(...)) -> or(x, ...) -- flatten
-  /// TODO: or(..., x, not(x)) -> or(..., '1) -- complement
-
   return {};
 }
 
+void OrOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
+                                       MLIRContext *context) {
+  struct Folder final : public OpRewritePattern<OrOp> {
+    using OpRewritePattern::OpRewritePattern;
+    LogicalResult matchAndRewrite(OrOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto inputs = op.inputs();
+      auto size = inputs.size();
+      APInt value;
+
+      if (size > 1) {
```

Comment:
nit, but I'd sink the "size > 1" check down into the match pattern line since it is specific to that check.

Reply:
I tend to remove this condition similar to the previous question.